src/core/PancakeInfinity.sol

```diff
@@ -508,22 +507,27 @@ abstract contract PancakeInfinity is SettlerAbstract {
                 if (uint256(poolManagerId) == 0) {
                     poolKey.poolManager = CL_MANAGER;
 
-                    swapParams.zeroForOne = zeroForOne;
-                    swapParams.amountSpecified = amountSpecified;
-                    // TODO: price limits
-                    swapParams.sqrtPriceLimitX96 = uint160(
-                        zeroForOne.ternary(
-                            uint160(4295128740), uint160(1461446703485210103287273052203988822378723970341)
-                        )
+                    delta = IPancakeInfinityCLPoolManager(address(poolKey.poolManager)).unsafeSwap(
+                        poolKey,
+                        IPancakeInfinityCLPoolManager.SwapParams({
+                            zeroForOne: zeroForOne,
+                            amountSpecified: amountSpecified,
+                            sqrtPriceLimitX96: uint160(
+                                zeroForOne.ternary(
+                                    uint160(4295128740), uint160(1461446703485210103287273052203988822378723970341)
+                                )
+                            )
+                        }),
+                        hookData
```

Comment:
Constructing the `SwapParams` here causes a memory leak. Don't allocate inside a loop. Can you make the `SwapParams` members arguments to `unsafeSwap`? (also can you restore the removed TODO)